django/db/models/sql/query.py

```diff
@@ -1121,6 +1121,8 @@ def build_filter(self, filter_expr, branch_negated=False, current_negated=False,
         query. However, if the filter isn't added to the query then the caller
         is responsible for unreffing the joins used.
         """
+        if isinstance(filter_expr, dict):
```

Comment:
A concern I have, looking at this, which I didn't explicitly mention in the original ticket [because I've never done such], is that this would allow the same accidental non-error if given an `OrderedDict`, because it supports the same expansion but isn't a subclass of the `dict` type.

I think I'm now reaching for real niche use-cases that have presumably never knowingly happened [or they'd have been reported long before I noticed], but still...